app/src/main/java/com/islaharper/dawnofandroid/domain/useCases/UseCases.kt

```diff
@@ -0,0 +1,9 @@
+package com.islaharper.dawnofandroid.domain.useCases
+
+import com.islaharper.dawnofandroid.domain.useCases.readOnboarding.ReadOnBoardingStateUseCase
+import com.islaharper.dawnofandroid.domain.useCases.saveOnboarding.SaveOnBoardingStateUseCase
+
+data class UseCases(
```

Comment:
Why use data class here? Do we expect to use `.copy` and `equals` ? 

Maybe we mean a Kotlin object:

```kotlin
object UseCases {
   val saveOnBoardingUseCase: SaveOnBoardingUseCase
}
```

or, why even have this class when we can just inject the use case we want using Dagger?

```kotlin
@Inject
class ViewModel(
   val saveOnBoardingUseCase: SaveOnBoardingUseCase, // injected using dagger
)
```

Reply:
I structured the UseCases this way as some ViewModels may require multiple use cases, so rather than injecting 3 or 4 separate use cases, it's injected once and all the required cases are accessible

Reply:
Hmm, I don't think this is correct from a design perspective. The idea of use cases is you can inject them as dependencies for a ViewModel and immediately look at the ViewModel and tell what it does (screaming architecture):

https://proandroiddev.com/why-you-need-use-cases-interactors-142e8a6fe576

Not only that, we use use cases to compose. So if there are two use cases that are used together (e.g., `LoginUseCase` and `SaveTokenUseCase`), we compose them somehow into one use case `LoginAndSaveTokenUseCase` and inejct that.

Bundling them together would also go against principle of least knowledge or at least the 'I` in interface segregation from SOLID.

Is bundling all the use cases together in a class a pattern you have seen somewhere else? If so, maybe you could provide a link or an alternative explanation so I can understand what's going on here.

Reply:
The bundling of Use Cases was the method shown to me at Glucode for a project we worked on there. After reading the above link and others I see that was not the best approach. Going forward I will separate use cases as well as use multiple methods within a use case if need.eg a SignIn usecase with methods for signInWithEmail and signInWithGoogle.